src/lib/diagnostics.ts

```diff
@@ -6,211 +6,185 @@
  */
 import {messages} from './messages';
 import * as vscode from 'vscode';
-import {Logger} from "./logger";
-import {TelemetryService} from "./external-services/telemetry-service";
-import * as targeting from './targeting';
-import * as Constants from './constants'
-
-type DiagnosticConvertibleLocation = {
-    file: string;
-    startLine: number;
-    startColumn: number;
+
+export type CodeLocation = {
+    // These all should be optional just like it is over at:
```

Comment:
This is great context! Is this worth keeping as a comment in code, or could we move it out? I think it's helpful today, but not sure it needs to persist past this refactor